### PR TITLE
Add teams administration

### PR DESCRIPTION
**Pull request by Remg**

### CODE REVIEW

File: src/Entity/Reporting/TeamMemberHistory.php [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+namespace App\Entity\Reporting;
+
+use App\Entity\Adherent;
+use App\Entity\Administrator;
+use App\Entity\Team\Team;
+use Doctrine\ORM\Mapping as ORM;
+
+/**
+ * @ORM\Table(
+ *     name="team_member_histories",
```
Comment (illusionOfParadise) [inline src/Entity/Reporting/TeamMemberHistory.php:12]: We used to apply the singular for the table name:
```suggestion
 *     name="team_member_history",
```

File: src/Entity/Reporting/TeamMemberHistory.php [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+    {
+        return $this->administrator;
+    }
+
+    public function getDate(): \DateTimeImmutable
+    {
+        return $this->date;
+    }
+
+    public function isAdd(): bool
```
Comment (illusionOfParadise) [inline src/Entity/Reporting/TeamMemberHistory.php:117]: What about ?
```suggestion
    public function isAdded(): bool
```
And the same for `isRemoved`?

File: src/Entity/Team/Member.php [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+use App\Entity\EntityTimestampableTrait;
+use Doctrine\ORM\Mapping as ORM;
+use Ramsey\Uuid\Uuid;
+use Ramsey\Uuid\UuidInterface;
+use Symfony\Bridge\Doctrine\Validator\Constraints\UniqueEntity;
+use Symfony\Component\Validator\Constraints as Assert;
+
+/**
+ * @ORM\Entity(repositoryClass="App\Repository\Team\MemberRepository")
+ * @ORM\Table(uniqueConstraints={
```
Comment (illusionOfParadise) [inline src/Entity/Team/Member.php:16]: What about naming the table as `team_member`?
```suggestion
 * @ORM\Table(name="team_member", uniqueConstraints={
```

File: src/Entity/Team/Member.php [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+    {
+        return (string) $this->adherent;
+    }
+
+    public function getTeam(): ?Team
+    {
+        return $this->team;
+    }
+
+    public function setTeam(?Team $team): void
```
Comment (illusionOfParadise) [inline src/Entity/Team/Member.php:66]: The property is not nullable.
```suggestion
    public function setTeam(Team $team): void
```
Same for `setAdherent`.

File: src/Entity/Team/Team.php [hunk 5]
```diff
@@ -0,0 +1,134 @@
...
+    {
+        return (string) $this->name;
+    }
+
+    public function getType(): ?string
+    {
+        return $this->type;
+    }
+
+    public function setType(?string $type): void
```
Comment (illusionOfParadise) [inline src/Entity/Team/Team.php:94]: The property `type` is not nullable.
```suggestion
    public function setType(string $type): void
```
Same for `setName`.

Comment (Remg) [reply src/Entity/Team/Team.php:94]: needed for the Form. validators will do its job.

File: config/services/admin.xml [hunk 6]
```diff
@@ -339,6 +339,30 @@
             <argument type="service" id="App\Filesystem\FileManager"/>
         </service>
 
+        <service id="app.admin.team" class="App\Admin\TeamAdmin">
```
Comment (ottaviano) [inline config/services/admin.xml:342]: what about moving these Admin classes into `App\Admin\Team` namespace (like entity classes)?

File: src/Admin/TeamAdmin.php [hunk 7]
```diff
@@ -0,0 +1,164 @@
...
+    protected function configureFormFields(FormMapper $formMapper)
+    {
+        $formMapper
+            ->with('Informations ⚙️', ['class' => 'col-md-6'])
+                ->add('type', ChoiceType::class, [
+                    'label' => 'Type',
+                    'choices' => TypeEnum::ALL,
+                    'choice_label' => function (string $choice) {
+                        return "team.type.$choice";
+                    },
```
Comment (ottaviano) [inline src/Admin/TeamAdmin.php:52]: we can create a FormType for `TeamTypeChoice` and re use it in `configureDatagridFilters`?

File: translations/messages.fr.yml [hunk 8]
```diff
@@ -1148,3 +1148,14 @@ scope.feature.mobile_app: Application Mobile
...
+#
+# Team
+#
+team.type.phoning: Phoning
+
+#
+# TeamMemberHistory
+#
+team_member_history.action.add: Ajout
+team_member_history.action.remove: SUppression
```
Comment (ottaviano) [inline translations/messages.fr.yml:1161]:
```suggestion
team_member_history.action.remove: Suppression
```

File: translations/validators.fr.yml [hunk 9]
```diff
@@ -512,3 +512,15 @@ internal_api_application.validation.application_name_length: Le nom de l'applica
 # AudienceSegment
 #
 audience_segment.scope.invalid_choice: Le scope n'est pas valide.
+
+#
+# Team
+#
+team.member.already_exists: Cet adhérent fait déjà parti de cette équipe.
```
Comment (ottaviano) [inline translations/validators.fr.yml:519]:
```suggestion
team.member.already_exists: Cet adhérent fait déjà partie de cette équipe.
```

File: translations/validators.fr.yml [hunk 10]
```diff
@@ -512,3 +512,15 @@ internal_api_application.validation.application_name_length: Le nom de l'applica
...
+team.member.adherent.not_blank: Veuillez sélectiionner un adhérent.
```
Comment (ottaviano) [inline translations/validators.fr.yml:520]:
```suggestion
team.member.adherent.not_blank: Veuillez sélectionner un adhérent.
```

File: translations/validators.fr.yml [hunk 11]
```diff
@@ -512,3 +512,15 @@ internal_api_application.validation.application_name_length: Le nom de l'applica
...
+team.type_name.already_exists: Une équipe du même type porte déjà le même nom.
+team.name.not_blank: Veuillez spécifier un nom d'équipe.
+team.name.min_length: Le nom de l'équipe doit fait au moins {{ limit }} caractères.
```
Comment (ottaviano) [inline translations/validators.fr.yml:523]:
```suggestion
team.name.min_length: Le nom de l'équipe doit faire au moins {{ limit }} caractères.
```

**Review by ottaviano — APPROVED** [review]

**Comment by ottaviano** [comment]
[EM-1190]

**Pull request merged**